libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -180,6 +181,14 @@ void CompilerUtils::encodeToMemory(
 		t = t->mobileType()->interfaceType(_encodeAsLibraryTypes)->encodingType();
 	}
 
+	if (_givenTypes.empty())
```

Comment:
The optimizer (if not the peephole optimizer) will fix that, but yes, I can try to extract it.